os/drivers/input/ist415.c

```diff
@@ -490,6 +496,10 @@ static void ist415_stop_device(struct ist415_dev_s *dev)
 {
 	ist415vdbg("%s\n", __func__);
 
+	while (sem_wait(&dev->sem) != OK) {
+		ASSERT(get_errno() == EINTR);
```

Comment:
It call ASSERT, not retry, is it right?

Reply:
yes it's checking errno == EINTR. sem_wait fail case must be EINTR only. If not system is corrupted